### ISSUE TITLE
MotorStorm [BCES00006] Some 2D Textures not transparent

### ISSUE THREAD

**Issue by captainh00ch** [RPCS3/rpcs3#10936]
The 2D textures for mud and fire are not transparent, and render as black chunks. It also on rare occations happen sporadically, but it's consistent when driving through mud and when the car explodes. The intensity seems to vary, But I'm suspecting it's because the problem is with the particle effects. It seems like instead of screening out the black parts of a 2D texture, it just renders the whole images. (Or something else, I'm not sure, that's just a guess.)

Screenshot of issue:

![msglitch](https://user-images.githubusercontent.com/91194747/134882924-11e875c7-f805-4ee4-824c-998b7143fa15.JPG)

I have tried all four shader modes, antialiasing auto/disabled, strict rendering, all zcull accuracies, default settings, nothing seems to do anything with the issue. I noticed another thread mentioning this back in 2019 [(source)](https://forums.rpcs3.net/thread-183124-post-305398.html#pid305398) , but I havent seen anything else about it, and most data from this game predates that. 
Other than this, the game runs very good, stable 30-ish fps on my surprisingly mediocre rig.

To reproduce, drive through mud, or blow up the car. The textures are flickering when they appear.

Specs:
Windows 10 Pro 20H2 (Build no. 19042.1237)
GeForce 1080
Ryzen 7 3700X @ 3.60 GHz
32GB DDR4 @ 1066MHz

RSX Captures (I took several, not sure which is the best):

https://www.mediafire.com/file/yr3uyucl3w64lzp/captures.zip/file

RenderDoc:
I have tried to get this glitch documented in RenderDoc, It just does not show up. I am 99.9% sure that the frames captured were the glitchy ones. Tossing it in, if it helps.

https://drive.google.com/file/d/1p9UE8kZuei84g6668CVcc6ERyC3PIbHt/view?usp=sharing

Log:
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/7234463/RPCS3.log)

**Comment by captainh00ch** [comment]
If there is anything else you need, let me know

**Comment by captainh00ch** [comment]
Okay, so I tried the game with OpenGL renderer, and the problem does not persist. But openGL performance is less than half the FPS of Vulkan.

**Comment by captainh00ch** [comment]
Oh shit sorry about the empty log. It should be the most recent RPCS3. I'm gonna see if I can boot and reproduce it tomorrow, and give you an updated log.

**Comment by legend800** [comment]
@kd-11 Yes, it's a definite regression caused by some change in the past month or so. 

I was able to get in game once today and grabbed this screen. 
![2021-11-01 23_40_56-Window](https://user-images.githubusercontent.com/20983844/139798908-ed13fdd7-6fe8-48d3-95f2-07b08ac45cf3.png)

Unfortunately, after that I can no longer get past the title screen like another user just reported here:
https://github.com/RPCS3/rpcs3/issues/8376#issuecomment-956500114

Here's the log for freeze:
[Freeze.gz](https://github.com/RPCS3/rpcs3/files/7458374/Freeze.gz)

To sum up, here's the full list of issues:

1. Freeze in menus - new regression
2. Black dirt (this issue) - new regression
3. Car stutter - https://github.com/RPCS3/rpcs3/issues/10936
4. Misc gfx issues - https://github.com/RPCS3/rpcs3/issues/8373

**Comment by kd-11** [comment]
I am unable to reproduce this issue on AMD and NV GPUs (pascal). If possible capture a before and after renderdoc trace of the issue on pascal or rdna2.

**Comment by legend800** [comment]
Once we can figure out why it's freezing/crashing in the menus, I can grab that. 

I've been testing the game for years on rpcs3 and have never encountered this. It's always at this exact spot (log above). I've tried various settings. This regression needs to be fixed or we need to find the new special settings it requires. 

![2021-11-03 10_01_20-Window](https://user-images.githubusercontent.com/20983844/140122216-45c50838-0c57-4886-81d2-f917d8f56911.png)

**Comment by Satan86** [comment]
I can run the Game fine and got a Renderdoc Capture with my 1050 Ti 
https://drive.google.com/file/d/1zi04e2uOVZzVBGgxxk78dqA2uJ0a_Tsd/view?usp=sharing

**Comment by legend800** [comment]
Fixed the menu crash - due to unrelated data corruption regressions around Trophies and temp save data files. 

Anyway, found that this commit broke the game - causing massive perf issues (fixed soon after) + black boxes as noted in this thread.
https://github.com/RPCS3/rpcs3/pull/10900

**Comment by kd-11** [comment]
Test if https://github.com/RPCS3/rpcs3/pull/11204 fixes this one.

**Comment by legend800** [comment]
Yep, all fixed, thank you! 

Not only did you fix the regression but the "purple dirt" is also fixed finally. Nice work!
